internal/resource/config/sp/idpconnection/sp_idp_connection_resource.go

```diff
@@ -0,0 +1,4529 @@
+										"name": schema.StringAttribute{
+											Required:            true,
+											Description:         "The name of this attribute.",
+											MarkdownDescription: "The name of this attribute.",
+										},
+									},
+								},
+								Optional:            true,
+								Description:         "A list of assertion attributes that are automatically populated by PingFederate.",
+								MarkdownDescription: "A list of assertion attributes that are automatically populated by PingFederate.",
+							},
+							"extended_attributes": schema.SetNestedAttribute{
+								NestedObject: schema.NestedAttributeObject{
+									Attributes: map[string]schema.Attribute{
+										"masked": schema.BoolAttribute{
+											Optional:            true,
+											Computed:            true,
+											Description:         "Specifies whether this attribute is masked in PingFederate logs. Defaults to false.",
+											MarkdownDescription: "Specifies whether this attribute is masked in PingFederate logs. Defaults to false.",
+											Default:             booldefault.StaticBool(false),
+										},
+										"name": schema.StringAttribute{
+											Required:            true,
+											Description:         "The name of this attribute.",
+											MarkdownDescription: "The name of this attribute.",
+										},
+									},
+								},
+								Optional:            true,
+								Description:         "A list of additional attributes that are receive in the incoming assertion.",
+								MarkdownDescription: "A list of additional attributes that are receive in the incoming assertion.",
+							},
+						},
+						Required:            true,
+						Description:         "A set of user attributes that this server will receive in the token.",
+						MarkdownDescription: "A set of user attributes that this server will receive in the token.",
+					},
+					"generate_local_token": schema.BoolAttribute{
+						Required:            true,
+						Description:         "Indicates whether a local token needs to be generated. The default value is false.",
+						MarkdownDescription: "Indicates whether a local token needs to be generated. The default value is false.",
+					},
+					"token_generator_mappings": schema.ListNestedAttribute{
+						NestedObject: schema.NestedAttributeObject{
+							Attributes: map[string]schema.Attribute{
+								"attribute_contract_fulfillment": schema.MapNestedAttribute{
+									NestedObject: schema.NestedAttributeObject{
+										Attributes: map[string]schema.Attribute{
+											"source": schema.SingleNestedAttribute{
+												Attributes: map[string]schema.Attribute{
+													"id": schema.StringAttribute{
+														Optional:            true,
+														Description:         "The attribute source ID that refers to the attribute source that this key references. In some resources, the ID is optional and will be ignored. In these cases the ID should be omitted. If the source type i
```

Comment:
You'll need to give these a default value, or there will be errors about the plugin returning an invalid type if AttributeSources ever is nil. Same with issuance criteria below